libvast/test/system/datagram_source.cpp

```diff
@@ -0,0 +1,103 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#define SUITE datagram_source
+
+#include "vast/system/datagram_source.hpp"
+
+#include "test.hpp"
+#include "test/data.hpp"
+
+#include "fixtures/actor_system_and_events.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <iterator>
+
+#include <caf/exit_reason.hpp>
+#include <caf/io/middleman.hpp>
+#include <caf/send.hpp>
+
+#include "vast/format/bro.hpp"
+
+using namespace vast;
+using namespace vast::system;
+
+namespace {
+
+struct test_sink_state {
+  std::vector<table_slice_handle> slices;
+  inline static constexpr const char* name = "test-sink";
+};
+
+using test_sink_type = caf::stateful_actor<test_sink_state>;
+
+caf::behavior test_sink(test_sink_type* self, caf::actor src) {
+  self->send(src, sink_atom::value, self);
+  return {
+    [=](caf::stream<table_slice_handle> in) {
+      return self->make_sink(
+        in,
+        [](caf::unit_t&) {
+          // nop
+        },
+        [=](caf::unit_t&, table_slice_handle ptr) {
+          self->state.slices.emplace_back(std::move(ptr));
+        },
+        [=](caf::unit_t&, const error&) {
+          CAF_MESSAGE(self->name() << " is done");
+        }
+      );
+    }
+  };
+}
+
+} // namespace <anonymous>
+
+FIXTURE_SCOPE(source_tests, fixtures::deterministic_actor_system_and_events)
+
+TEST(bro ssl source) {
```

Comment:
@Neverlord Did you mean `s/ssl/udp/`?

Reply:
No, I've originally intended to use the bro SSL logs. The test should be named `bro conn source` instead, since it's using conn logs now.

Reply:
Okay, I renamed it.